src/runtime/metal/metal_device_api.mm

```diff
@@ -55,8 +55,13 @@
         break;
       }
       case kWarpSize: {
-        // Set warp size to be 1 for safty reason.
+#if defined(__x86_64__)
         *rv = 1;
+#elif defined(__aarch64__)
+        *rv = 32;
+#else
+        LOG(FATAL) << "Unknown architecture";
```

Comment:
as a fall back, perhaps we should use `LOG(WARNING)` saying it's not x86 or aarch64, and then return 1 as a fallback

Reply:
i agree @jinhongyii do you might do that?